animal-sort/animalSort.js

```diff
@@ -0,0 +1,21 @@
+function animalSort (collection, iteratees = []) {
+  if (collection === null) {
+    return []
```

Comment:
Should return `null`.
If the method is called with `null`, they will get an empty array. Sorting shouldn't change the type.